src/library_wasmfs_fetch.js

```diff
@@ -38,8 +38,12 @@ mergeInto(LibraryManager.library, {
         }
       }
       var response = await fetch(url);
-      var buffer = await response['arrayBuffer']();
-      wasmFS$JSMemoryFiles[file] = new Uint8Array(buffer);
+      if (response.ok) {
+        var buffer = await response['arrayBuffer']();
+        wasmFS$JSMemoryFiles[file] = new Uint8Array(buffer);
+      } else {
+        throw (response.status);
```

Comment:
```suggestion
        throw response.status;
```

Reply:
Actually, perhaps we should do this:
```suggestion
        throw response;
```
And then check for `.status` in the catch. That might be clearer than throwing just the status code itself without saying what it is.

Reply:
Yes, I agree, we can get more information from a response.